### PR TITLE
add custom_plugin_auth_enabled_field

### PR DESCRIPTION
**Pull request by jeffy-mathew**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3927687672)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: b4afcff59c9e8f1172ccf607ae2c4042a56c6454 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3927709586)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 25d4872fb1884699aa0346de6c35eeffc4c7bb01 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3927851362)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: a498d7a6e4e148a24a11f8060a0645b47c7dc17e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3929527791)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 3a82bf9bf687c8525cb91a85c737a4ccd0783d99 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3930044354)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 267d32b53580cd936b24e22555fb732ea3e45d99 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3930056794)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: a42e00851a2e80acd13365c61a90ea73a021fcf7 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3930097268)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 8b4a5e2cbcc6dd7a4072ac58e3e03c1b12e0827b 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3930127311)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 3151ab3d8b0d6204ae2e65237de0eee10b7ac442 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3932696231)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: d3aa8a31dc86e09ba9306af03e5e9cdfe43f43ec 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3933009661)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: c487e05b67feecaffe6622efb744f4b663e567d8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3933120804)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 0b9ba248f576cbf79d1f3670a55d1b35019c59e8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3935494374)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 5e6e4cc9b63096f9b06064ad1cd5fe96b61ddc16 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936181969)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: fcb406de8b813a2f994af1ef86c9b1285dec6b1a 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936239671)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 99c08d3804c346e4e0caaeef619e7638ab845494 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936283242)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4647/merge
Commit: e67c7953336b9e601586c3f511a43d93f9707cc2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936531121)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 7f9a9ec62199d04f7f639ea945ff08d98d17fe20 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936703649)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: a831c05be373bb2f4cb7c94df32545678522b568 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3936701271)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 91b96bd3d0b9b0a5e1e177690cdc75723354e228 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3937764629)

File: apidef/schema.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ const Schema = `{
         },
         "enable_coprocess_auth": {
             "type": "boolean"
+        },
+		"custom_plugin_auth_enabled": {
```
Comment (titpetric) [inline apidef/schema.go:66]: Can you file an issue to move the schema into a .json file and an embedfs?

File: bin/ci-tests.sh [hunk 2]
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
 
+# print a command and execute it
+show() {
```
Comment (titpetric) [inline bin/ci-tests.sh:4]: Revert this please, bash has `set -x` to echo commands, and `set +x` to stop. The changes in this file are not required, lets leave CI improvements for separate PRs, so we don't modify release code and test code at the same time

File: gateway/api_loader.go [hunk 3]
```diff
@@ -360,37 +360,35 @@ func (gw *Gateway) processSpec(spec *APISpec, apisByListen map[string]int,
...
+					&authArray,
+					&GoPluginMiddleware{
+						BaseMiddleware: baseMid,
+						Path:           mwAuthCheckFunc.Path,
+						SymbolName:     mwAuthCheckFunc.Name,
+						APILevel:       true,
+					},
+				)
+			default:
+				// TODO: check if mwAuthCheckFunc is available/valid
```
Comment (titpetric) [inline gateway/api_loader.go:386]: Don't leave TODOs in code, resolve them

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: 666188428b84814119872a44d5810c2f464e59c5 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3938446277)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: b062c87fbbe23928ae990cb64e16a8cd307f2a90 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3938617999)

File: apidef/schema.go (hunk 1 above)
Comment (jeffy-mathew) [reply apidef/schema.go:66]: yes

Comment (jeffy-mathew) [reply apidef/schema.go:66]: https://tyktech.atlassian.net/browse/TT-7617

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4647/merge
Commit: cc23c0f8ff68f20f19d28de2d287a46bd058513d 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3938923773)

**Review by titpetric — APPROVED** [review]

**Pull request merged**